discovery_engine_core/web-api/src/models.rs

```diff
@@ -50,8 +63,64 @@ pub(crate) enum Error {
 impl Reject for Error {}
 
 /// A unique identifier of a document.
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Display, AsRef)]
-pub(crate) struct DocumentId(pub(crate) String);
+#[derive(
+    AsRef,
+    Clone,
+    Debug,
+    Display,
+    PartialEq,
+    Eq,
+    Hash,
+    Serialize,
+    Deserialize,
+    sqlx::Type,
+    sqlx::FromRow,
+)]
+#[sqlx(transparent)]
+pub struct DocumentId(String);
+
+impl DocumentId {
+    pub(crate) fn new(id: impl Into<String>) -> Result<Self, Error> {
+        let id = id.into();
+
+        if id.is_empty() {
+            Err(Error::DocumentIdEmpty)
+        } else if id.contains('\u{0000}') {
+            Err(Error::DocumentIdContainsNul)
+        } else {
+            Ok(Self(id))
+        }
+    }
+}
+
+#[derive(Clone, Debug, Display, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct DocumentPropertyId(String);
```

Comment:
@acrrd mentioned that we should check this for nul char and empty, same as `UserId` and `DocumentId`

Reply:
in this pr we only retrieve and construct `DocumentPropertyId`s from our internal database, we never perform these checks in that case as we assume that we stored them correctly in the first place.

in the case where we have to validate `DocumentPropertyId`s coming from the outside, the code is added in the follow up pr #646, otherwise i had to mark it as dead code here.

Reply:
sure, got it